rllib/algorithms/impala/impala.py

```diff
@@ -938,35 +938,38 @@ def learn_on_processed_samples(self) -> ResultDict:
             Aggregated results from the learner group after an update is completed.
 
         """
-        result = {}
-        # There are batches on the queue -> Send them to the learner group.
+        # There are batches on the queue -> Send them all to the learner group.
         if self.batches_to_place_on_learner:
-            batch = self.batches_to_place_on_learner.pop(0)
+            batches = self.batches_to_place_on_learner[:]
+            self.batches_to_place_on_learner.clear()
```

Comment:
I don't personally trust `batches = self.batches_to_place_on_learner[:]`. can we do a copy operation to make sure it's not passing things by reference?

Reply:
We can also purge them at the very end and to do 
```
for batch in self.batches_to_place_on_learner:
     pass

self.batches_to_place_on_learner.clear()

```

Reply:
I actually don't understand your concern. The `[:]` operator creates a shallow copy of the list (each item is copied - by reference, not deeply - to the new list).
Clearing the original list afterwards will leave the original list empty and the new list will carry the (still valid) items' references.

Reply:
can we just use copy.copy() for shallow copy to be explicit? I don't know how slicing is implemented under the hood :)